src/datachain/lib/dc.py

```diff
@@ -446,7 +457,24 @@ def from_dataset(
             chain = DataChain.from_dataset("my_cats")
             ```
         """
-        return DataChain(name=name, version=version, session=session, settings=settings)
+        query = DatasetQuery(
+            name=name,
+            version=version,
+            session=session,
+            indexing_column_types=File._datachain_column_types,
+        )
+        telemetry.send_event_once("class", "datachain_init", name=name, version=version)
```

Comment:
do we need to put it now into all other methods (e.g. from_json?)

Reply:
Every `from_*` method ends up calling this, so I guess not.

Reply:
Note that I'm not actually changing the logic, I'm just making the existing one more obvious. We should probably rethink it eventually but I guess it's good enough for now.